main/lsp/test/error_reporter_test.cc

```diff
@@ -55,297 +55,313 @@ TEST_CASE("NotifiesVSCodeWhenFileHasErrors") {
+    auto lastDiagnosticLatency = move(counterStateDatabase.getTimings("last_diagnostic_latency").front());
+    auto lastDiagnosticDuration = lastDiagnosticLatency->end.usec - lastDiagnosticLatency->start.usec;
+    CHECK_GT(chrono::microseconds(lastDiagnosticDuration), chrono::milliseconds(10));
 }
 
 TEST_CASE("FirstAndLastLatencyAboutEqualWhenNoErrors") {
     auto gs = makeGS();
     auto cs = makeConfig();
     ErrorReporter er(cs);
+    auto epoch = 0;
+    auto newEpoch = 1;
+    auto file = make_shared<core::File>("foo.rb", "foo", core::File::Type::Normal, epoch);
+    core::FileRef fref;
     {
         core::UnfreezeFileTable fileTableAccess(*gs);
-        auto epoch = 0;
-        auto file = make_shared<core::File>("foo.rb", "foo", core::File::Type::Normal, epoch);
-        auto fref = gs->enterFile(file);
-        auto outputVector = dynamic_pointer_cast<LSPOutputToVector>(cs->output);
-
-        vector<unique_ptr<core::Error>> errors;
-        errors.emplace_back(
-            make_unique<core::Error>(core::Loc(fref, 0, 0), core::ErrorClass{1, core::StrictLevel::True}, "MyError",
-                                     vector<core::ErrorSection>(), vector<core::AutocorrectSuggestion>(), false));
-
-        vector<unique_ptr<Timer>> diagnosticLatencyTimers;
-        diagnosticLatencyTimers.emplace_back(make_unique<Timer>(logger, "last_diagnostic_latency"));
-
-        er.beginEpoch(epoch, move(diagnosticLatencyTimers));
-        Timer::timedSleep(chrono::milliseconds(5), *logger, "debugging");
-        er.pushDiagnostics(epoch, fref, errors, *gs);
-        Timer::timedSleep(chrono::milliseconds(5), *logger, "debugging");
-        er.endEpoch(epoch);
-
-        auto counters = getAndClearThreadCounters();
-        auto counterStateDatabase = CounterStateDatabase(move(counters));
-        auto firstDiagnosticLatency = move(counterStateDatabase.getTimings("first_diagnostic_latency").front());
-        auto firstDiagnosticDuration = firstDiagnosticLatency->end.usec - firstDiagnosticLatency->start.usec;
-        auto lastDiagnosticLatency = move(counterStateDatabase.getTimings("last_diagnostic_latency").front());
-        auto lastDiagnosticDuration = lastDiagnosticLatency->end.usec - lastDiagnosticLatency->start.usec;
-
-        INFO("first_ and last_diagnostic_latency ~equal when there are no errors to report");
-        CHECK_LT(chrono::microseconds(firstDiagnosticDuration), chrono::milliseconds(10));
-        CHECK_LT(chrono::microseconds(lastDiagnosticDuration), chrono::milliseconds(10));
+        fref = gs->enterFile(file);
     }
+
+    vector<unique_ptr<core::Error>> emptyErrorList;
+
+    vector<unique_ptr<Timer>> diagnosticLatencyTimers;
+    diagnosticLatencyTimers.emplace_back(make_unique<Timer>(logger, "last_diagnostic_latency"));
+    er.beginEpoch(epoch, move(diagnosticLatencyTimers));
+    Timer::timedSleep(chrono::milliseconds(5), *logger, "delay so timer is reported");
```

Comment:
Nit: Since we don't care about these timers getting reported we can elide the sleeps for this first epoch.

Reply:
🤦  I just realized that we probably don't need this first epoch since files _begin_ with an empty list.